lib/awful/spawn.lua

```diff
@@ -178,21 +178,75 @@ function spawn.with_line_callback(cmd, callbacks)
     return pid
 end
 
---- Asynchronously spawn a program and capture its output.
--- (wraps `spawn.with_line_callback`).
+--- Asynchronously spawn a program and (optionally) execute callbacks.
+--
+-- `cmd` is being run and in case `cb`, `cb_error` or `cb_signal` are provided
+-- they will be run, according to the exit code of `cmd`.
+--
+-- This wraps `spawn.with_line_callback`.
+--
 -- @tparam string|table cmd The command.
--- @tab callback Function with the following arguments
--- @tparam string callback.stdout Output on stdout.
--- @tparam string callback.stderr Output on stderr.
--- @tparam string callback.exitreason Exit Reason.
--- The reason can be "exit" or "signal".
--- @tparam integer callback.exitcode Exit code.
--- For "exit" reason it's the exit code.
--- For "signal" reason — the signal causing process termination.
--- @treturn[1] Integer the PID of the forked process.
+--
+-- @tparam[opt] function cb Function to be called on success.
+-- @tparam string cb.stdout The output on stdout.
+-- @tparam string cb.stderr The output on stderr.
+--
+-- @tparam[opt] function cb_error Callback on errors.
+--   By default a message will be printed to awesome's stderr using
+--   `gears.debug.print_error`.
+-- @tparam string cb_error.stdout The output on stdout.
+-- @tparam string cb_error.stderr The output on stderr.
+-- @tparam string cb_error.exitcode The exit code (will be != 0).
+-- @tparam[opt='error'] string cb_signal.type 'error' always
+--
+-- @tparam[opt] function cb_signal Callback on signals.
+--   By default a message will be printed to awesome's stderr using
+--   `gears.debug.print_error`.
+-- @tparam string cb_signal.stdout The output on stdout.
+-- @tparam string cb_signal.stderr The output on stderr.
+-- @tparam string cb_signal.signal The signal.
```

Comment:
`The signal` is a bit unspecific, I think. Signal number? Signal name? Should people be pointed to `awesome.signal_I_forgot_the_exact_name` that allows to map between signal numbers and names?